### PR TITLE
Simplify more

### PR DESCRIPTION
**Pull request by mnoack**
**WHY** Whilst setting up CreditLink which uses this gem, it wasn't clear what was wrong, so I've improved error handling, and in doing simplified the classes so verifiers don't exist, we simply have Local and Remove configurations which know how to verify themselves.

### CODE REVIEW

File: lib/pansophy_authenticator/configuration/remote.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module PansophyAuthenticator
+  module Configuration
+    class Remote < Base
+      def verification
+        Result.new errors
+      end
+
+      def errors
+        errors = super
+        errors << 'Bucket name is not defined' if @bucket_name.nil?
```
Comment (AlessandroBerardi) [inline lib/pansophy_authenticator/configuration/remote.rb]: This is destructive, as oppose as `+` you used in local.

File: lib/pansophy_authenticator/configuration/base.rb [hunk 2]
```diff
@@ -3,32 +3,26 @@
...
   module Configuration
-    class Instance
-      include Anima.new :local, :bucket_name, :file_path, :application, :cache_store
+    class Base
+      include Anima.new :bucket_name, :file_path, :application, :cache_store
       include Memoizable
 
       def local?
-        @local
+        is_a?(Local)
```
Comment (alxberardi) [inline lib/pansophy_authenticator/configuration/base.rb]: Expecting a parent class to know about its child classes is an anti-pattern imho

**Comment by alxberardi** [comment]
👍

**Comment by mnoack** [comment]
:+1:

**Comment by alvinypyim** [comment]
👍

**Pull request merged**